crates/ruma-events/src/room/join_rules.rs

```diff
@@ -17,12 +24,17 @@ pub struct JoinRulesEventContent {
     /// The type of rules used for users wishing to join this room.
     #[ruma_event(skip_redaction)]
     pub join_rule: JoinRule,
+
+    /// Allow rules used for the `restricted` join rule.
+    #[serde(default)]
+    #[ruma_event(skip_redaction)]
+    pub allow: Vec<AllowRule>,
 }
 
 impl JoinRulesEventContent {
     /// Creates a new `JoinRulesEventContent` with the given rule.
-    pub fn new(join_rule: JoinRule) -> Self {
-        Self { join_rule }
+    pub fn new(join_rule: JoinRule, allow: Vec<AllowRule>) -> Self {
```

Comment:
Either way this constructor should be left as-is though. If `allow` is kept as a field of `JoinRulesEventContent`, it should just be initialized as an empty `Vec`.

Reply:
Would it make sense to have a `restricted` constructor for `JoinRuleEventContent` like
```rust
fn restricted(allow: Vec<AllowRule>) -> Self {
    Self { join_rule: JoinRule::Restricted, allow }
}
```
If it's kept as a field?

Reply:
Sure 🙂